### PR TITLE
fix: dropIndexes so it run on the primary regardless the read preference MONGOSH-1478

### PR DESCRIPTION
**Pull request by kmruiz**

### CODE REVIEW

File: packages/shell-api/src/database.ts [hunk 1]
```diff
@@ -26,13 +26,14 @@ import {
...
-  WriteConcern,
-  ListCollectionsOptions,
+import {
+  type ChangeStreamOptions,
+  type CommandOperationOptions,
+  type CreateCollectionOptions,
+  type Document,
+  type WriteConcern,
+  type ListCollectionsOptions,
+  ReadPreference,
```
Comment (addaleax) [inline packages/shell-api/src/database.ts:36]: Where is `ReadPreference` being used? Generally speaking, we don’t want to import runtime values from the Node.js driver via `@mongosh/service-provider-core`, only types

File: packages/shell-api/src/database.ts [hunk 2]
```diff
@@ -149,10 +150,25 @@ export default class Database extends ShellApiWithMongoClass {
...
-
   public async _runCommand(
     cmd: Document,
     options: CommandOperationOptions = {}
+  ): Promise<Document> {
+    return this._mongo._serviceProvider.runCommandWithCheck(
+      this._name,
+      adjustRunCommand(cmd, this._instanceState.shellBson),
+      {
+        readPreference: 'primaryPreferred',
```
Comment (addaleax) [inline packages/shell-api/src/database.ts:161]: I would imagine that we don’t want to apply any read preference here at all, to match what drivers and (presumably) the legacy shell do?

File: packages/shell-api/src/database.ts (hunk 1 above)
Comment (kmruiz) [reply packages/shell-api/src/database.ts:36]: Ah yes, you are totally right, this is a leftover. Fixing it!

File: packages/shell-api/src/database.ts (hunk 2 above)
Comment (kmruiz) [reply packages/shell-api/src/database.ts:161]: OK, will fix everything in the same commit as the previous left over.

**Review by addaleax — APPROVED** [review]
Sorry, forgot to actually approve this! Feel free to ping me when this happens again :)

**Review by addaleax — APPROVED** [review]

**Pull request merged**